### PR TITLE
[llvm-mca][AArch64] Add AArch64 version of clearsSuperRegisters.

### PR DESCRIPTION
**Pull request by Asher8118**
This patch overrides the clearsSuperRegisters method defined in MCInstrAnalysis to identify register writes that clear the upper portion of all super-registers on AArch64 architecture.

On AArch64, a write to a general-purpose register of 32-bit data size is defined to use the lower 32-bits of the register and zero extend the upper 32-bits.
Similarly, SIMD and FP instructions operating on scalar data only access the lower bits of the SIMD&FP register. The unused upper bits are cleared to zero on a write.
This also applies to SIMD vector registers when the element size in bits multiplied by the number of lanes is lower than 128. The upper 64 bits of the vector register are cleared to zero on a write.

### CODE REVIEW

File: llvm/lib/Target/AArch64/MCTargetDesc/AArch64MCTargetDesc.cpp
Comment (mshockwave) [inline llvm/lib/Target/AArch64/MCTargetDesc/AArch64MCTargetDesc.cpp:462]:
```suggestion
      return FPR8RC.contains(RegID) || FPR16RC.contains(RegID) ||
             FPR32RC.contains(RegID) || FPR64RC.contains(RegID)
```

**Review by mshockwave — COMMENTED** [review]
LGTM w/ a minor comment. But I would leave to the ARM folks to approve.

File: llvm/lib/Target/AArch64/MCTargetDesc/AArch64MCTargetDesc.cpp
Comment (Rin18) [reply llvm/lib/Target/AArch64/MCTargetDesc/AArch64MCTargetDesc.cpp]: Done

File: llvm/test/tools/llvm-mca/AArch64/Neoverse/V2-clear-upper-regs.s
Comment (peterwaller-arm) [inline llvm/test/tools/llvm-mca/AArch64/Neoverse/V2-clear-upper-regs.s:17]: I think this could have a FPR64 (`ldr d0`) and FPR128 (`ldr q0`) test, the latter of which also looks like it needs implementing.

**Review by peterwaller-arm — COMMENTED** [review]
I found a minor thing which looks missing, but once that's fixed this has my approval.

File: llvm/test/tools/llvm-mca/AArch64/Neoverse/V2-clear-upper-regs.s
Comment (Rin18) [reply llvm/test/tools/llvm-mca/AArch64/Neoverse/V2-clear-upper-regs.s:17]: Thanks for pointing this out, done

**Review by peterwaller-arm — APPROVED** [review]

**Pull request merged**
